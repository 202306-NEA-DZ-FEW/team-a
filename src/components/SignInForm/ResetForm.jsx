import { useFormik } from "formik";
import * as Yup from "yup";

import { useAuth } from "@/context/AuthProvider";

import Input from "../Input";

function ResetForm({ t }) {
  const { resetPassword } = useAuth();
  const resetPassForm = useFormik({
    initialValues: {
      resetEmail: "",
    },
    validationSchema: Yup.object({
      resetEmail: Yup.string()
        .email(t("signIn:emailValid"))
        .required(t("signIn:emailRequired")),
    }),
    onSubmit: (values) => {
      resetPassword(values.resetEmail);
    },
  });
  return (
    <dialog id='my_modal_3' className='modal'>
      <div className='modal-box'>
        <h3 className='font-bold text-lg'>{t("signIn:EnterYourEmail")}</h3>
        <form onSubmit={resetPassForm.handleSubmit}>
          <Input
            name='resetEmail'
            type='email'
            placeholder={t("signIn:emailPlaceholder")}
            label={t("signIn:emailLabel")}
            handleChange={resetPassForm.handleChange}
            value={resetPassForm.values.resetEmail}
            handleBlur={resetPassForm.handleBlur}
            error={resetPassForm.errors.resetEmail}
            touched={resetPassForm.touched.resetEmail}
          />
          <button type='submit' className='btn btn-sm btn-info my-4'>
            {t("signIn:sendMeResetLink")}
          </button>
        </form>
      </div>
      <form method='dialog' className='modal-backdrop'>
        <button>close</button>
      </form>
    </dialog>
  );
}

export default ResetForm;
