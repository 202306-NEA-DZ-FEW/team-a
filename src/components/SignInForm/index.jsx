import { useFormik } from "formik";
import Link from "next/link";
import { BsFacebook, BsGithub } from "react-icons/bs";
import { FcGoogle } from "react-icons/fc";
import * as Yup from "yup";

import { useAuth } from "@/context/AuthProvider";

import ResetForm from "./ResetForm";
import Checkbox from "../Checkbox";
import Input from "../Input";

function SignInForm({ t }) {
  const { signIn, signInWithFacebook, signInWithGoogle, signInWithGithub } =
    useAuth();
  const formik = useFormik({
    initialValues: {
      email: "",
      password: "",
      save: false,
    },
    validationSchema: Yup.object({
      email: Yup.string()
        .email(t("signIn:emailValid"))
        .required(t("signIn:emailRequired")),
      password: Yup.string().required(t("signIn:passwordRequired")),
      resetPassword: Yup.string().email(t("signIn:emailValid")),
    }),
    onSubmit: (values) => {
      const email = values.email;
      const password = values.password;
      signIn(email, password);
    },
  });

  return (
    <section className='flex gap-2 flex-col w-full mx-auto'>
      <h1 className='text-3xl font-bold mb-8'>{t("signIn:signIn")}</h1>
      {/* Form Section */}
      <form className='flex flex-col gap-2' onSubmit={formik.handleSubmit}>
        <Input
          name='email'
          type='email'
          placeholder={t("signIn:emailPlaceholder")}
          label={t("signIn:emailLabel")}
          handleChange={formik.handleChange}
          value={formik.values.email}
          handleBlur={formik.handleBlur}
          error={formik.errors.email}
          touched={formik.touched.email}
        />
        <Input
          name='password'
          type='password'
          placeholder={t("signIn:PasswordPlaceholder")}
          label={t("signIn:passwordLabel")}
          handleChange={formik.handleChange}
          value={formik.values.password}
          handleBlur={formik.handleBlur}
          error={formik.errors.password}
          touched={formik.touched.password}
        />
        {/* Checkbox & Password Forgotten Section */}
        <div className='flex flex-col md:flex-row justify-between md:items-center flex-wrap'>
          <Checkbox
            name='save'
            label={t("signIn:savedlogin")}
            handleChange={formik.handleChange}
            value={formik.values.terms}
            handleBlur={formik.handleBlur}
            error={formik.errors.terms}
            touched={formik.touched.terms}
          />
          <button
            onClick={() => document.getElementById("my_modal_3").showModal()}
            className='btn btn-link link-error font-normal hover:font-bold self-start btn-sm normal-case'
          >
            {t("signIn:passwordForgotten")}
          </button>
        </div>
        {/* Login Button */}
        <button
          className='btn btn-secondary text-black bg-opacity-40 w-full normal-case text-xl font-normal self-center rounded-xl '
          type='submit'
          disabled={formik.isSubmitting}
        >
          {formik.isSubmitting ? t("signIn:loading") : t("signIn:signInButton")}
        </button>
      </form>
      <span className='divider'>{t("signIn:divider")}</span>
      {/* Social Medias Section */}

      <p className='text-center my-4'>{t("signIn:signUpMethod")}</p>
      <div className='flex gap-2 justify-center items-center mb-4'>
        <button
          className='btn btn-ghost rounded-xl hover:drop-shadow-xl transition-all duration-400 ease-in-out'
          onClick={signInWithFacebook}
        >
          <BsFacebook className='text-3xl text-blue-600' />
        </button>
        <button
          onClick={signInWithGoogle}
          className='btn btn-ghost rounded-xl hover:drop-shadow-xl transition-all duration-400 ease-in-out'
        >
          <FcGoogle className='text-3xl' />
        </button>
        <button
          onClick={signInWithGithub}
          className='btn btn-ghost rounded-xl hover:drop-shadow-xl transition-all duration-400 ease-in-out'
        >
          <BsGithub className='text-3xl' />
        </button>
      </div>
      <div className='flex flex-col md:flex-row justify-center items-center'>
        <p className='text-base'>{t("signIn:noAccount")}</p>
        <Link href='/auth/sign-up' className='btn btn-link'>
          {t("signIn:signUp")}
        </Link>
      </div>
      {/* Password Reset Modal */}
      <ResetForm t={t} />
    </section>
  );
}

export default SignInForm;
