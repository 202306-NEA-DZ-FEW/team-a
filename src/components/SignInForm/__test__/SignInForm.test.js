import renderer from "react-test-renderer";

import SignInForm from "..";

it("renders correctly", () => {
  const mockT = jest.fn();
  const tree = renderer.create(<SignInForm t={mockT} />).toJSON();
  expect(tree).toMatchSnapshot();
});
