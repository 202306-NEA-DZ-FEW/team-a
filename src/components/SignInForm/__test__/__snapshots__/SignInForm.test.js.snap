// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="flex flex-col w-full px-20 lg:px-14"
>
  <h1
    className="text-3xl font-bold text-center md:mt-0 mt-10"
  />
  <form
    className="flex flex-col gap-2"
    onSubmit={[Function]}
  >
    <div>
      <div
        className="label ml-2"
      >
        <label
          className="label-text"
          htmlFor="email"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <input
        className="input bg-white input-primary w-full rounded-full border-opacity-25"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <div
        className="label ml-2"
      >
        <label
          className="label-text"
          htmlFor="password"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <input
        className="input bg-white input-primary w-full rounded-full border-opacity-25"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="flex flex-col md:flex-row justify-around gap-4 md:gap-8 items-center my-2"
    >
      <div>
        <div
          className="flex items-center gap-2"
        >
          <input
            className="checkbox checkbox-primary "
            id="save"
            name="save"
            onBlur={[Function]}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="label label-text"
            htmlFor="save"
          />
        </div>
      </div>
      <button
        className="btn btn-link btn-sm p-0 block capitalize md:mb-0 mb-4"
        onClick={[Function]}
      />
    </div>
    <button
      className="btn btn-primary w-full self-center rounded-3xl text-white md:mb-0 mb-8"
      disabled={false}
      type="submit"
    />
  </form>
  <span
    className="mt-4 mb-0 divider"
  />
  <div>
    <p
      className="text-center font-light my-4"
    />
    <div
      className="flex gap-4 justify-center items-center md:mb-0 mb-10"
    >
      <button
        className="btn btn-circle btn-sm bg-white"
      >
        <svg
          className="text-3xl text-blue-600"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
          />
        </svg>
      </button>
      <button
        className="btn btn-circle btn-sm bg-white"
      >
        <svg
          className="text-3xl"
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 48 48"
          width="1em"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#FFC107"
          />
          <path
            d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
            fill="#FF3D00"
          />
          <path
            d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
            fill="#4CAF50"
          />
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#1976D2"
          />
        </svg>
      </button>
      <button
        className="btn btn-circle btn-sm bg-white"
      >
        <svg
          className="text-3xl text-gray-700"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="mt-4 flex flex-col md:flex-row justify-center items-center gap-2"
  >
    <p
      className="text-base"
    />
    <a
      className="btn btn-link"
      href="/auth/sign-up"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
  </div>
  <dialog
    className="modal"
    id="my_modal_3"
  >
    <div
      className="modal-box"
    >
      <h3
        className="font-bold text-lg"
      />
      <form
        onSubmit={[Function]}
      >
        <div>
          <div
            className="label ml-2"
          >
            <label
              className="label-text"
              htmlFor="resetEmail"
            />
            <label
              className="label-text-alt "
            />
          </div>
          <input
            className="input bg-white input-primary w-full rounded-full border-opacity-25"
            id="resetEmail"
            name="resetEmail"
            onBlur={[Function]}
            onChange={[Function]}
            type="email"
            value=""
          />
        </div>
        <button
          className="btn btn-md btn-info w-full self-center my-8 rounded-full"
          type="submit"
        />
      </form>
    </div>
    <form
      className="modal-backdrop"
      method="dialog"
    >
      <button>
        close
      </button>
    </form>
  </dialog>
</section>
`;
